packages/battery_drivers/battery.py

```diff
@@ -98,13 +98,28 @@ def shutdown(self):
 
     def turn_off(self, timeout: int = 20, wait: bool = False, callback: Optional[Callable] = None):
         #   This is a battery shutdown, the power will be cut off after `timeout` seconds
-        timeout_str = f'{timeout}'.zfill(2)
-        self._interaction = BatteryInteraction(
-            name="turn_off",
-            command=f'Q{timeout_str}'.encode('utf-8'),
-            check=lambda d: d.get('TTL(sec)', None) == timeout,
-            callback=callback
-        )
+
+        firmware_version = self.info.get("firmware_version")
+        # multi-firmware support
+        # TODO: create classes as version-handlers, parent for common-handlers
+        if firmware_version == "v2.0.0":
+            timeout_str = f'{timeout}'.zfill(2)
+            self._interaction = BatteryInteraction(
+                name="turn_off",
+                command=f'Q{timeout_str}'.encode('utf-8'),
+                check=lambda d: d.get('TTL(sec)', None) == timeout,
+                callback=callback,
+            )
+        elif firmware_version == "v2.0.1":
```

Comment:
when we release a new version, this will break. Use `semver` to work with versions instead (https://pypi.org/project/semver/). Any version v2.0.1+ should go with this version of the shutdown command.

Reply:
Comment addressed :smiley: Please have a look again. Also tested on both battery versions.

Reply:
Nice!